### PR TITLE
Improve SemanticConvention generation template

### PR DESCRIPTION
**Pull request by Sherlouk**
Based off of #276, will need to change the base once that work is complete. 
**Please just review the second commit for now.**

At the moment, if you run the `generate` script, it will make changes to the project which will cause compiler or test failures.

The three failures which this protects against:
* The value "int" being used as an enum value. This needs to be "Int" instead.
* The keyword "internal" being used as a case value. This should be wrapped with backticks like `internal`.
* The Swift telemetry language which we have chosen to add, outside of the specification.

While not technically a failure, this change will also revert the documentation changes made in a previous PR which changed all the `Int` values to "int" incorrectly. This fixes that too.

### CODE REVIEW

**Comment by Sherlouk** [comment]
Added a second commit here which adds better support for enum values. Instead of just saying the type should be String, it says which enum value should be used. It will even add a clickable link to the enum - so users can more easily see the available options.

In theory, we could also list them in the inline docs - but lets iterate and see how it feels.

File: Scripts/semantic-convention/templates/SemanticAttributes.swift.j2 [hunk 1]
```diff
@@ -76,14 +83,21 @@ public enum {{enum}}: String {
     /**
     {{attribute.brief | to_doc_brief}}.
     */
+    /// - Tag: otel{{class_name}}
```
Comment (Sherlouk) [inline Scripts/semantic-convention/templates/SemanticAttributes.swift.j2:86]: I have absolutely no idea why we must use triple slash syntax for tags, but everything else works with `/**`.

Tags do not work unless you use triple slash syntax.
I have verified this functionality, both work side by side.

**Comment by nachoBonafonte** [comment]
Everything looks really great, I only have some doubt with this:
> The Swift telemetry language which we have chosen to add, outside of the specification.

The value was added to the spec in this [PR](https://github.com/open-telemetry/opentelemetry-specification/pull/2097). So it should be included in the next version of the spec. We will probably only use the script for updating when a newer version appear. What do you think?

**Comment by Sherlouk** [comment]
Noted, will remove the Swift workaround and will rebase onto main.

**Comment by Sherlouk** [comment]
Should all be good now 👍

**Review by nachoBonafonte — APPROVED** [review]
Yes, it looks perfect now. Thanks a lot

**Comment by Sherlouk** [comment]
Thanks Nacho!

**Pull request merged**